realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -331,6 +335,7 @@ struct JStringCharsAccessor {
     const jstring m_string;
     const jchar* const m_data;
     const size_t m_size;
+    const bool m_owned;
```

Comment:
This name is a bit confusing to me. Owned by who? Looking at the code, it seems to imply "owned by this class" so maybe "owned_by_accessor" or "delete_on_free" or pr. the comment above, just remove it? 🤔

Reply:
See other comments about the rationale. I can update the name to `delete_on_free` or maybe `delete_jstring_ref_on_free` to be very explicit.

Reply:
I'm fine with keeping this. Either of the tow names you propose there seems better to me than owned.